File: Benchmark-Application/benchmarks/CreareThread.cpp
#include <iostream>
#include <thread>
#include <vector>
#include <ctime>
#include <numeric>

using namespace std;
using namespace chrono;

const int NUMBER_THREADS = 10000;

void threadFunction(int& result) {
}

void exec() {
    // Record the start time
    high_resolution_clock::time_point start_time = high_resolution_clock::now();

    vector<thread> threads;
    vector<int> results(NUMBER_THREADS, 0);

    for (int i = 0; i < NUMBER_THREADS; ++i) {
        threads.emplace_back(threadFunction, ref(results[i]));
    }

    for (auto& thread : threads) {
        thread.join();
    }

    // Calculate the elapsed time in seconds
    high_resolution_clock::time_point end_time = high_resolution_clock::now();
    duration<double> elapsed_time = duration_cast<duration<double>>(end_time - start_time);

    cout << scientific << elapsed_time.count();
}

int main() {
    exec();

    return 0;
}

File: Benchmark-Application/benchmarks/CreareThread.py
import threading
import time
from datetime import datetime

NUMBER_THREADS = 10000


def thread_function(result_list):
    pass


def exec_program():
    # Record the start time
    start_time = time.time()

    threads = []
    results = [0] * NUMBER_THREADS

    for i in range(NUMBER_THREADS):
        t = threading.Thread(target=thread_function, args=(results,))
        threads.append(t)
        t.start()

    for t in threads:
        t.join()

    # Calculate the elapsed time in seconds
    end_time = time.time()
    elapsed_time = end_time - start_time

    print(f"{elapsed_time:.2e}")


if __name__ == "__main__":
    exec_program()
